sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -1035,7 +1034,11 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
             m.appendTail(${termResult});
             ${ev.primitive} = ${classNameUTF8String}.fromString(${termResult}.toString());
             ${ev.isNull} = false;
+          } else {
+            ${ev.isNull} = true;
```

Comment:
For UnaryExpression, it's better to use `ev.isNull = eval.isNull`, for others, using `ev.isNull = true` will be better.